src/main/java/com/codeless/api/automation/service/impl/EmailServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.codeless.api.automation.service.impl;
+
+import com.codeless.api.automation.entity.User;
+import com.codeless.api.automation.service.EmailService;
+import com.codeless.api.automation.util.VerificationUtil;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class EmailServiceImpl implements EmailService {
+
+  private final JavaMailSender javaMailSender;
+
+  @Override
+  public void sendEmail(User user, HttpServletRequest httpServletRequest) {
+    MimeMessage mimeMessage = generateVerificationMessage(user, httpServletRequest);
+    log.info("Sending email message to : '{}' ", user.getUsername());
+    javaMailSender.send(mimeMessage);
+  }
+
+  public MimeMessage generateVerificationMessage(User user, HttpServletRequest httpServletRequest) {
+    MimeMessage mimeMessage = javaMailSender.createMimeMessage();
+    try {
+      MimeMessageHelper mimeMessageHelper = new MimeMessageHelper(mimeMessage, true, "utf-8");
+      mimeMessageHelper.setTo(user.getUsername());
+      mimeMessageHelper.setSubject("Codeless Automation Registration verification");
+      mimeMessageHelper.setFrom("codelessautomation@gmail.com");
+
+      String verificationToken = VerificationUtil.createVerificationToken(user);
+      String baseUrl = ServletUriComponentsBuilder.fromRequestUri(httpServletRequest)
+          .replacePath("codeless")
+          .build()
+          .toUriString();
+      String verificationUrl = baseUrl + "/verify?verification-token=" + verificationToken;
+
```

Comment:
Please use template for rendering email.

if you need reference, it could be found here src/main/java/com/codeless/api/automation/notification/EmailServiceImpl.java

Reply:
update with template usage for email rendering